tests/chainer_tests/functions_tests/array_tests/test_pad_sequence.py

```diff
@@ -0,0 +1,75 @@
+import unittest
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import functions
+from chainer import gradient_check
+from chainer import testing
+from chainer.testing import attr
+
+
+@testing.parameterize(*testing.product({
+    'lengths': [[2, 1, 5, 3], [2], [0]],
+    'length': [None, 6, 5],
```

Comment:
Instead of explicit max length (`5`), how about specifying it as `'max'` for example, and rewrite it in `setUp()` like this?

```
if self.length == 'max':
    self.length = max(self.lengths)
```

This way, the extra corner case `lengths=[0]`, `length=0` will be also covered.